COMETwebapp/ViewModels/Components/ReferenceData/CategoriesTableViewModel.cs

```diff
@@ -217,5 +233,134 @@ private void UpdateRows(IEnumerable<CategoryRowViewModel> rowsToDisplay)
                 this.Rows.Items.First(x => x.Category.Iid == existingRow.Category.Iid).UpdateProperties(existingRow);
             }
         }
+
+        /// <summary>
+        /// Method invoked when canceling the deprecation/un-deprecation of a <see cref="Category"/>
+        /// </summary>
+        public void OnCancelButtonClick()
+        {
+            this.popupVisible = false;
+        }
+
+        /// <summary>
+        /// Method invoked when confirming the deprecation/un-deprecation of a <see cref="Category"/>
+        /// </summary>
+        public async void OnConfirmButtonClick()
+        {
+            if (this.Category.IsDeprecated)
+            {
+                await this.UnDeprecatingCategory();
+            }
+            else
+            {
+                await this.DeprecatingCategory();
+            }
+            popupVisible = false;
+        }
+
+        /// <summary>
+        /// Action invoked when the deprecate or undeprecate button is clicked
+        /// </summary>
+        public void OnDeprecateUnDeprecateButtonClick(GridCommandColumnCellDisplayTemplateContext context)
```

Comment:
Missing documentation. Also, provide a CategoryRowViewModel as parameter and not the GridCommandColumnCellDisplay. That type is DevExpress related, VM should not be dependent of UI framework